frontend/app/src/model/task.ts

```diff
@@ -43,7 +43,7 @@ export function taskCompletion<R, M extends TaskMeta>(
       (actionResult, meta) => {
         taskManager.unregisterHandler(task, taskId);
         const { result, message } = actionResult;
-        if (message) {
+        if (message && result === null) {
```

Comment:
actually in theory (I don't think we don't populate the message in error but in theory it may happen) message can be an empty string and result being in a null for an error.

Reply:
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
error with empty string doesn't make sense right?

Reply:
It's the default value. It can happen.

Reply:
Btw this has been working for all the `async_query` tasks so far (only with the message checking) so at least in the existing endpoints it doesn't seem to be a problem or we would have noticed something obviously wrong.

Reply:
I am telling you what can happen with the way the API is written. I don't know how much clearer I can be. It's possible to have a `result == null` and `msg == ""`, which is a completely unhelpful error, but still an error.

Reply:
Issue has been opened to think about this #2731